src/main/java/org/opensearch/security/action/apitokens/ApiTokenAction.java

```diff
@@ -418,8 +426,15 @@ void validateUserPermissions(List<String> clusterPermissions, List<ApiToken.Inde
                         List<String> rolePatterns = indexPermission.getIndex_patterns();
                         List<String> roleIndexPerms = indexPermission.getAllowed_actions();
 
+                        IndexPattern indexPattern = IndexPattern.from(rolePatterns);
+
                         // Check if this role's pattern covers the requested pattern
-                        if (WildcardMatcher.from(rolePatterns).test(requestedPattern)) {
+                        if (indexPattern.matches(
+                            requestedPattern,
+                            indexNameExpressionResolver,
+                            (String template) -> WildcardMatcher.NONE,
```

Comment:
I am a bit confused about returning `WildcardMatcher.NONE` here. Why do we do this?

Reply:
Since I changed the index pattern class to take in a matcher generator, here I need to pass in a function to generate a wildcardmatcher (as I don't have a PrivilegesEvaluationContext to pass in). In the Index pattern class, the wildcardmatcher is used to evaluate pattern templates/user attributes, which we don't plan to support for API tokens, and thus I passed in a WildCardMatcher that doesn't match anything to not support that use case.